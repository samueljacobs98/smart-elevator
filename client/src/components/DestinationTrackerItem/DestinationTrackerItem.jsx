import "./DestinationTrackerItem.scss";

const DestinationTrackerItem = ({ floor, lifts, arrived }) => {
  return (
    <div key={floor} className="item">
      <p className="item__box item__box--floor">{floor}</p>
      <div className="item__box item__box--lifts">
        {lifts.map((lift) => {
          const modifier = arrived.has(lift) ? "item__box__lift--arrived" : "";
          return (
            <p key={lift} className={"item__box__lift" + modifier}>
              {lift}
            </p>
          );
        })}
      </div>
    </div>
  );
};

export default DestinationTrackerItem;
