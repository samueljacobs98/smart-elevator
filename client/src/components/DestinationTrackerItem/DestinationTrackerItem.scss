@use "../../assets/scss/variables" as *;

.item {
  display: flex;

  &__box {
    width: 100%;
    padding: 0.5rem;
    display: flex;
    align-items: center;

    &--floor {
      max-width: 3rem;
      background: $colour-gradient-linear-green;
      justify-content: center;
    }

    &--lifts {
      background: $colour-gradient-linear-blue;
      justify-content: flex-start;
      gap: 0.5rem;
      flex-wrap: wrap;
    }

    &__lift {
      &--arrived {
        color: $colour-yellow;
        font-weight: $font-weight-bold;
      }
    }
  }
}
