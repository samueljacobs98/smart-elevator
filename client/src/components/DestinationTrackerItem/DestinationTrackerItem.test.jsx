import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import DestinationTrackerItem from "./DestinationTrackerItem";

const floor = 5;
const lifts = ["A", "B", "C"];
const arrived = new Set(["A", "C"]);

describe("DestinationTrackerItem", () => {
  const renderComponent = () =>
    render(
      <DestinationTrackerItem floor={floor} lifts={lifts} arrived={arrived} />
    );

  it("renders without crashing", () => {
    renderComponent();

    expect(screen).toBeDefined();
  });

  it("displays the correct floor number", () => {
    renderComponent();

    const floorNumber = screen.getByText(floor);

    expect(floorNumber).toBeInTheDocument();
  });

  it("renders the correct number of lifts", () => {
    renderComponent();

    lifts.forEach((lift) => {
      const liftElement = screen.getByText(lift);

      expect(liftElement).toBeInTheDocument();
    });
  });

  it("applies the arrived modifier to the className of the arrived lifts", () => {
    renderComponent();

    lifts.forEach((lift) => {
      const liftElement = screen.getByText(lift);

      let expectedClassName = "item__box__lift";

      if (arrived.has(lift)) {
        expectedClassName += " item__box__lift--arrived";
      }

      expect(liftElement).toHaveClass(expectedClassName);
    });
  });

  it("should match snapshot", () => {
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
  });
});
