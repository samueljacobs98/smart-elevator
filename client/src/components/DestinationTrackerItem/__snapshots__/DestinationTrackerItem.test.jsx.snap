// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DestinationTrackerItem should match snapshot 1`] = `
<div>
  <div
    class="item"
  >
    <p
      class="item__box item__box--floor"
    >
      5
    </p>
    <div
      class="item__box item__box--lifts"
    >
      <p
        class="item__box__lift item__box__lift--arrived"
      >
        A
      </p>
      <p
        class="item__box__lift "
      >
        B
      </p>
      <p
        class="item__box__lift item__box__lift--arrived"
      >
        C
      </p>
    </div>
  </div>
</div>
`;
